x-pack/plugins/case/server/connectors/types.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from 'kibana/server';
+import {
+  ActionTypeConfig,
+  ActionTypeSecrets,
+  ActionTypeParams,
+  ActionType,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../actions/server/types';
+import { CaseResponse, ConnectorTypes } from '../../common/api';
+import { CaseClientGetAlertsResponse } from '../client/alerts/types';
+import {
+  CaseServiceSetup,
+  CaseConfigureServiceSetup,
+  CaseUserActionServiceSetup,
+  ConnectorMappingsServiceSetup,
+  AlertServiceContract,
+} from '../services';
+
+export interface GetActionTypeParams {
+  logger: Logger;
+  caseService: CaseServiceSetup;
+  caseConfigureService: CaseConfigureServiceSetup;
+  connectorMappingsService: ConnectorMappingsServiceSetup;
+  userActionService: CaseUserActionServiceSetup;
+  alertsService: AlertServiceContract;
+}
+
+export interface RegisterConnectorsArgs extends GetActionTypeParams {
+  actionsRegisterType<
+    Config extends ActionTypeConfig = ActionTypeConfig,
+    Secrets extends ActionTypeSecrets = ActionTypeSecrets,
+    Params extends ActionTypeParams = ActionTypeParams,
+    ExecutorResultData = void
+  >(
+    actionType: ActionType<Config, Secrets, Params, ExecutorResultData>
+  ): void;
+}
+
+export type FormatterConnectorTypes = Exclude<ConnectorTypes, ConnectorTypes.none>;
+
+export interface ExternalServiceFormatter<TExternalServiceParams = {}> {
+  format: (
+    theCase: CaseResponse,
+    alerts: CaseClientGetAlertsResponse
+  ) => Promise<TExternalServiceParams>;
```

Comment:
I did not see/understand why format need to return a Promise since we are not doing any async request inside of the format function.

Reply:
I make it async in case in the feature we need to do an async operation inside it. But you are right not needed at the moment!